### PR TITLE
Deep Lake mini upgrades

### PR DESCRIPTION
**Pull request by davidbuniat**
Improvements
* set default num_workers for ingestion to 0
* upgraded notebooks for avoiding dataset creation ambiguity
* added `force_delete_dataset_by_path`
* bumped deeplake to 3.3.0
* creds arg passing to deeplake object that would allow custom S3

Notes
* please double check if poetry is not messed up (thanks!)

Asks
* Would be great to create a shared slack channel for quick questions

### CODE REVIEW

File: docs/use_cases/code/twitter-the-algorithm-analysis-deeplake.ipynb [hunk 1]
```diff
@@ -120,7 +120,9 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "db = DeepLake.from_documents(texts, embeddings, dataset_path=\"hub://davitbun/twitter-algorithm\")"
+    "username = \"davitbun\" # replace with your username from app.activeloop.ai\n",
+    "db = DeepLake(dataset_path=f\"hub://{username}/twitter-algorithm-4\", embedding_function=embeddings, public=True) #dataset owuld be public\n",
```
Comment (aiswaryasankar) [inline docs/use_cases/code/twitter-the-algorithm-analysis-deeplake.ipynb:124]: Nit: typo `dataset wuld be public`

File: tests/integration_tests/vectorstores/test_deeplake.py [hunk 2]
```diff
@@ -164,3 +164,9 @@ def test_delete_dataset_by_filter(deeplake_datastore: DeepLake) -> None:
     assert len(deeplake_datastore.ds) == 2
 
     deeplake_datastore.delete_dataset()
+
+
+def test_delete_by_path(deeplake_datastore: DeepLake) -> None:
+    """Test delete dataset."""
+    path = deeplake_datastore.dataset_path
+    DeepLake.force_delete_by_path(path)
```
Comment (aiswaryasankar) [inline tests/integration_tests/vectorstores/test_deeplake.py:172]: Perhaps add an assert to check it was successful?

File: docs/use_cases/code/twitter-the-algorithm-analysis-deeplake.ipynb (hunk 1 above)
Comment (davidbuniat) [reply docs/use_cases/code/twitter-the-algorithm-analysis-deeplake.ipynb:124]: thanks a lot for spotting it!

File: tests/integration_tests/vectorstores/test_deeplake.py (hunk 2 above)
Comment (davidbuniat) [reply tests/integration_tests/vectorstores/test_deeplake.py:172]: yep great idea!

**Review by aiswaryasankar — COMMENTED** [review]
Left a few small comments - overall question though, I see that this adds the overwrite=True as the default option but does it address the concern of hitting `Please use a url that points to an existing Deep Lake Dataset or an empty folder. If you wish to delete the folder and its contents, you may run deeplake.delete(dataset_path, force=True)` when the dataset upload failed for some reason.  I think probably another fix might be needed to fix that issue.

**Review by hwchase17 — APPROVED** [review]
thanks!

**Pull request merged**